### PR TITLE
Ignore git error on repository root detection.

### PR DESCRIPTION
**Pull request by isoos**
Needed for https://github.com/dart-lang/pub-dev/pull/8173

### CODE REVIEW

File: lib/src/package_analyzer.dart [hunk 1]
```diff
@@ -242,12 +242,16 @@ class PackageAnalyzer {
...
-    workingDirectory: packageDir,
-  );
-  if (pr.exitCode == 0) {
-    return pr.stdout.asString.trim();
+  try {
+    final pr = await runGitIsolated(
+      ['rev-parse', '--show-toplevel'],
+      workingDirectory: packageDir,
+    );
+    if (pr.exitCode == 0) {
```
Comment (sigurdm) [inline lib/src/package_analyzer.dart:250]: Can this ever be false?

File: lib/src/package_analyzer.dart [hunk 2]
```diff
@@ -242,12 +242,16 @@ class PackageAnalyzer {
...
+      return pr.stdout.asString.trim();
+    }
+  } on GitToolException catch (_) {
+    // ignore exception
```
Comment (sigurdm) [inline lib/src/package_analyzer.dart:254]:
```suggestion
    // not in a git directory (or git is broken) - ignore exception
```

**Review by sigurdm — APPROVED** [review]
LGTM with suggestions

**Review by sigurdm — APPROVED** [review]

File: lib/src/package_analyzer.dart (hunk 1 above)
Comment (isoos) [reply lib/src/package_analyzer.dart]: No, it is always true. I've removed that check.

**Pull request merged**